### ISSUE TITLE
Stability issues

### ISSUE THREAD

**Issue by de-gozaru** [naver/roma#6]
Hi,

Thank you for the great library.

I'm using the function `special_procrustes` to compute rotation matrices during training, and I noticed that this made the training a little bit unstable, and I suspect this is due to backpropagating through SVD.

Is there a way to obtain a mask that indicates the rotation matrices that are obtained via an unstable SVD, so that I will ignore them during the computation of the loss, and hence make the training more stable?

Thank you in advance for your help!

**Comment by rbregier** [comment]
Hi,

Thank you for the feedback.
Indeed, for some rare inputs, solutions to the Procrustes problem are not unique, and gradients are not well defined.

One could detect such configurations by analyzing the eigenvalues of the SVD decomposition. I will try to look into that but I don't have much bandwidth to do it currently.

In the meantime, you could try to play with the extra parameters of the function to regularize the backpropagation: `regularization` tends to make the input more similar to a rotation matrix, and  `gradient_eps` should help with exploding gradient issues.
Let me know if it helps.

**Comment by rbregier** [comment]
It would typically depend on the range of values of your training loss among other things, so it's hard to tell.

Maybe you could start with something like (1e-4, 1e-2) for training loss values typically equal to 1.

**Comment by rbregier** [comment]
Hi,
In the latest version of RoMa (1.3.3), it is now possible to retrieve the singular values of the SVD decomposition, which should enable fancy regularizations.
You can install it from GitHub,  I will wait a couple of days before updating the pip version to ensure stability.
Please let me know if you experiment with it, I am always curious about uses cases and I would be happy to get feedback (or help if needed).
